src/main/java/duke/enums/Command.java

```diff
@@ -0,0 +1,236 @@
+                String message = MessageFormat.format(template, "done");
+                throw new DukeException(message);
+            }
+        }
+    },
+
+    /**
+     * Adds an {@code Event} task to the {@code TaskManager}.
+     */
+    EVENT {
+        /**
+         * Validates whether the user input is of the correct format for the 'event' command.
+         *
+         * @param input the user input.
+         * @throws DukeException if the user input is invalid.
+         */
+        @Override
+        public void validate(String input) throws DukeException {
+            String regex = "^(?i)event\\s+.*\\S+.*\\s+/at\\s+.*\\S+.*\\s*$";
+            if (!Pattern.matches(regex, input)) {
+                String template = String.format("%s\n%s", ResourceHandler.getString("exception.invalidArgs"),
+                        ResourceHandler.getString("command.eventFormat"));
+                String message = MessageFormat.format(template, "event");
+                throw new DukeException(message);
+            }
+        }
+    },
+
+    /**
+     * Searches for {@code Task}s in the {@code TaskManager} which match the keyword.
+     */
+    FIND {
+        /**
+         * Validates whether the user input is of the correct format for the 'find' command.
+         *
+         * @param input the user input.
+         * @throws DukeException if the user input is invalid.
+         */
+        @Override
+        public void validate(String input) throws DukeException {
+            String regex = "^(?i)find\\s+.*\\S+.*\\s*$";
+            if (!Pattern.matches(regex, input)) {
+                String template = String.format("%s\n%s", ResourceHandler.getString("exception.invalidArgs"),
+                        ResourceHandler.getString("command.findFormat"));
+                String message = MessageFormat.format(template, "find");
+                throw new DukeException(message);
+            }
+        }
+    },
+
+    /**
+     * Lists all {@code Task}s in the {@code TaskManager}.
+     */
+    LIST {
+        /**
+         * Validates whether the user input is of the correct format for the 'list' command.
+         *
+         * @param input the user input.
+         * @throws DukeException if the user input is invalid.
+         */
+        @Override
+        public void validate(String input) throws DukeException {
+            String regex = "^(?i)list\\s*$";
+            if (!Pattern.matches(regex, input)) {
+                String template = ResourceHandler.getString("exception.noArgs");
+                String message = MessageFormat.format(template, "list");
+                throw new DukeException(message);
+            }
+        }
+    },
+
+    /**
+     * Lists all overdue {@code Task}s in the {@code TaskManager}.
+     */
+    OVERDUE {
+        /**
+         * Validates whether the user input is of the correct format for the 'overdue' command.
+         *
+         * @param input the user input.
```

Comment:
Perhaps you might want to consider putting Cap for the first letter of the description. For example: The user input

Reply:
Hmm, the reason I didn't do so is because the CS2103T coding standard doesn't explicitly list this as necessary (although the example does capitalise the first letter of the parameter descriptions). I've been mostly following the Java standard library JavaDoc format which does not practise capitalisation.

Reply:
Oooo ok. thanks for the explanation:)